server/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByScrollRequest.java

```diff
@@ -365,7 +365,8 @@ public Self setScroll(TimeValue keepAlive) {
      * Get scroll timeout
      */
     public TimeValue getScrollTime() {
-        return searchRequest.scroll().keepAlive();
+        Scroll scroll = searchRequest.scroll();
+        return scroll != null ? scroll.keepAlive() : null;
```

Comment:
Above would also remove the need for this change.

Reply:
I would suggest to keep it as scroll is nullable and there could be something calling this from the executor as well